plugins/hic/src/HicRenderer/HicRenderer.ts

```diff
@@ -89,6 +94,11 @@ export default class HicRenderer extends ServerSideRendererType {
           baseColor,
         ])
         ctx.fillRect((bin1 - offset) * w, (bin2 - offset) * w, w, w)
+        if (+Date.now() - start > 400) {
```

Comment:
the date is being used to count for the timeout?

Reply:
ya, the idea is to check every 400ms for whether an abort has arrived. The hic track can on take up to 5s to do a complete render so it intermittently checks to see if it should cancel

Reply:
the real trickyness was making sure the abort signal was passed through to the backend. I'm not sure if I have added enough tests to ensure this works but there are a couple added in the BaseRpcDriver...

Reply:
gotcha. Now the abort signal is getting passed as a prop?

Reply:
yep, actually every renderer will get a signal prop it just depends on whether they use it

Reply:
it would be valuable to use the lessons from this PR to spread it throughout the codebase

Reply:
I see, I see.